encodings/fastlanes/src/delta/compute.rs

```diff
@@ -1,5 +1,422 @@
-use vortex::compute::ArrayCompute;
+use std::cmp::min;
+
+use vortex::compute::unary::ScalarAtFn;
+use vortex::compute::{ArrayCompute, SliceFn};
+use vortex::{Array, IntoArray, IntoArrayVariant};
+use vortex_error::{vortex_bail, vortex_err, VortexExpect, VortexResult};
+use vortex_scalar::Scalar;
 
 use crate::DeltaArray;
 
-impl ArrayCompute for DeltaArray {}
+impl ArrayCompute for DeltaArray {
+    fn scalar_at(&self) -> Option<&dyn ScalarAtFn> {
+        Some(self)
+    }
+
+    fn slice(&self) -> Option<&dyn SliceFn> {
+        Some(self)
+    }
+}
+
+impl ScalarAtFn for DeltaArray {
+    fn scalar_at(&self, index: usize) -> VortexResult<Scalar> {
+        let decompressed = SliceFn::slice(self, index, index + 1)?.into_primitive()?;
```

Comment:
just `slice(self, index, index+1)` is there a reason you want to do `self.slice(index, index+1)`?

Reply:
hm fair, I haven't thought about arraycompute and individual functions colliding in scope. Fair to keep it as is